mlir-sycl/lib/Transforms/Inliner.cpp

```diff
@@ -309,16 +626,18 @@ void Inliner::collectCallOps(const CallGraphNode &SrcNode, const CallGraph &CG,
   });
 }
 
+bool InlinerBase::isRecursiveCall(const ResolvedCall &ResolvedCall) const {
+  return ResolvedCall.TgtNode->getCallableRegion()->isAncestor(
+      ResolvedCall.Call->getParentRegion());
+}
+
 //===----------------------------------------------------------------------===//
 // AlwaysInliner
 //===----------------------------------------------------------------------===//
 
-/// Returns true if the given call should be inlined.
-bool AlwaysInliner::shouldInline(ResolvedCall &ResolvedCall) const {
-  // Don't allow inlining if the target is an ancestor of the call. This
-  // prevents inlining recursively.
-  if (ResolvedCall.TgtNode->getCallableRegion()->isAncestor(
-          ResolvedCall.Call->getParentRegion()))
+bool AlwaysInliner::shouldInline(ResolvedCall &ResolvedCall,
+                                 Optional<CGUseList> Uses) const {
```

Comment:
No need to be optional, as it is always passed.
```suggestion
                                 CGUseList Uses) const {
```